core/src/main/java/io/grpc/LoadBalancer2.java

```diff
@@ -317,6 +322,77 @@ public String toString() {
   }
 
   /**
+   * Provides arguments for a {@link SubchannelPicker#pickSubchannel(PickSubchannelArgs)}.
+   */
+  public abstract static class PickSubchannelArgs {
```

Comment:
`PickSubchannelArgs` is public API, thus it should be in `io.grpc` (preferably as inner of `LoadBalancer2`) instead of `io.grpc.internal`.

Builder pattern exposes the construction of `PickSubchannelArgs` as public API, which is not needed by LB impls because they are only consumers of `PickSubchannelArgs`. The builder API has to define required fields (headers) and enforce it, but there isn't an ideal way to do it (there was a [previous discussion](https://github.com/grpc/grpc-java/pull/2667#discussion_r98262481) on a similar issue). It's better to avoid exposing such a builder unless it's necessary.